functions/notify_slack.py

```diff
@@ -260,34 +283,44 @@ def notify_slack(subject, message, region):
-        notification = iam_notification(message, region)
-        payload['text'] = "AWS IAM notification - " + message["detail-type"]
-        payload['attachments'].append(notification)
-    elif ("source" in message and message['source'] == "aws.iot"):
-        notification = iot_notification(message, region)
-        payload['text'] = "AWS Iot notification - " + message["detail-type"]
-        payload['attachments'].append(notification)
-    elif ("source" in message and message['source'] == "deployment"):
-        notification = deployment_notification(message, region)
-        payload['text'] = "AWS Deployment - " + message["detail-type"]
-        payload['attachments'].append(notification)
+    elif "source" in message:
+        if (message['source'] == "aws.ecs"):
+            notification = ecs_notification(message, region)
+            payload['text'] = "AWS ECS notification - " + message["detail-type"]
+            payload['attachments'].append(notification)
+        elif (message['source'] == "aws.ec2"):
+            notification = ectwo_notification(message, region)
+            payload['text'] = "AWS EC2 notification - " + message["detail-type"]
+            payload['attachments'].append(notification)
+        elif (message['source'] == "aws.rds"):
+            notification = rds_notification(message, region)
+            payload['text'] = "AWS RDS notification - " + message["detail-type"]
+            payload['attachments'].append(notification)
+        elif (message['Event Source'] in ["db-instance", "db-security-group", "db-parameter-group", "db-snapshot", "db-cluster", "db-cluster-snapshot"]):
+            notification = rds_event_subscription_notification(message, region)
+            payload['text'] = "AWS RDS notification - " + message["Event Message"]
+            payload['attachments'].append(notification)
+        elif (message['source'] == "aws.iam"):
+            notification = iam_notification(message, region)
+            payload['text'] = "AWS IAM notification - " + message["detail-type"]
+            payload['attachments'].append(notification)
+        elif (message['source'] == "aws.iot"):
+            notification = iot_notification(message, region)
+            payload['text'] = "AWS Iot notification - " + message["detail-type"]
+            payload['attachments'].append(notification)
+        elif (message['source'] == "deployment"):
+            notification = deployment_notification(message, region)
+            payload['text'] = "AWS Deployment - " + message["detail-type"]
+            payload['attachments'].append(notification)
+    elif "Origin" in message:
+        if (message['Origin'] or message['Destination'] == "AutoScalingGroup"):
+            notification = asg_notification(message, region)
+            if ("Terminating" in message.get('Description', "")):
+                event_type = "terminating"
+            elif ("Launching" in message.get('Description', "")):
```

Comment:
you may want a final else case where we just use the Description unformatted, I think if we get a [EC2 Instance Terminate Unsuccessful message ](https://docs.aws.amazon.com/autoscaling/ec2/userguide/cloud-watch-events.html#terminate-unsuccessful) this lambda may error. 🤔

Reply:
Yes, thank you for catching that!

Reply:
Added else statement and verified we will still get an unformatted post: https://smartrentcom.slack.com/archives/CK5A63GUU/p1624552901141800